### PR TITLE
[WIP] Fix Todo Issues

### PR DESCRIPTION
**Pull request by james-prado**
Closes more todo issues https://github.com/MyEtherWallet/MyEtherWallet/issues/519

- Add regex check for BIP44 paths
- Fix swap dropdown button height
- Remove `common/config` from ignored paths in jest_config

### CODE REVIEW

File: common/components/WalletDecrypt/components/Keystore.tsx [hunk 1]
```diff
@@ -1,34 +1,36 @@
...
+    return isKeystoreValid(file);
+  }
 
   public render() {
-    const { file, password } = this.props.value;
-    const passReq = isPassRequired(file);
-    const unlockDisabled = !file || (passReq && !password);
+    const { file, password, valid } = this.props.value;
+    const passReq = isKeystorePassRequired(file);
+    const unlockDisabled = !file || !valid || (passReq && !password);
```
Comment (dternyak) [inline common/components/WalletDecrypt/components/Keystore.tsx]: variablize

File: common/libs/validators.ts [hunk 2]
```diff
@@ -116,9 +116,8 @@ export function isPositiveIntegerOrZero(num: number): boolean {
 // https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki
 // normal path length is 4, ledger is the exception at 3
 export function isValidPath(dPath: string) {
```
Comment (dternyak) [inline common/libs/validators.ts]: could use a test. Where is the regex coming from? Handrolled?

File: common/libs/wallet/non-deterministic/helpers.ts [hunk 3]
```diff
@@ -33,22 +33,33 @@ export const signWrapper = (walletToWrap: IFullWallet): WrappedWallet =>
...
+      return KeystoreTypes.v1Encrypted;
+    } else if (parsed.hash && parsed.locked === false) {
+      return KeystoreTypes.v1Unencrypted;
+    } else if (parsed.publisher === 'MyEtherWallet') {
+      return KeystoreTypes.v2Unencrypted;
+    } else {
+      return '';
+    }
+  } catch {
+    return '';
```
Comment (dternyak) [inline common/libs/wallet/non-deterministic/helpers.ts]: Do we really want to return an empty string? What's the situation in which that's the desired return value?

File: common/libs/wallet/non-deterministic/helpers.ts [hunk 4]
```diff
@@ -33,22 +33,33 @@ export const signWrapper = (walletToWrap: IFullWallet): WrappedWallet =>
...
   }
 }
 
+const isKeystoreValid = (file: string): boolean => {
+  if (determineKeystoreType(file) === '') {
```
Comment (dternyak) [inline common/libs/wallet/non-deterministic/helpers.ts]: Ah, it's here. This seems quite unintuitive though.

File: common/libs/validators.ts (hunk 2 above)
Comment (james-prado) [reply common/libs/validators.ts]: Hand rolled, I'll provide some tests and some comments to clarify

File: common/libs/wallet/non-deterministic/helpers.ts (hunk 3 above)
Comment (james-prado) [reply common/libs/wallet/non-deterministic/helpers.ts]: The function is intended to return a string. Returning an empty string is falsey and cuts down on the amount of `try catch` statements if I just threw an error in the final `else {}` statement

Comment (dternyak) [reply common/libs/wallet/non-deterministic/helpers.ts]: @wbobeirne Maybe you have some thoughts on how we could make this more intuitive (or confirm you agree this is the best approach).

File: common/libs/wallet/non-deterministic/helpers.ts (hunk 4 above)
Comment (james-prado) [reply common/libs/wallet/non-deterministic/helpers.ts]: Yea it isn't very intuitive, the error handling still could use some polish. Errors thrown by parsing an invalid file as JSON in `determineKeystoreType` make it difficult to write something easier to use. I spent some time on it the issue but didn't make much progress. I don't think it's a blocking issue, and it is still functional, so I'm going to come back to it later after I've fixed more critical issues.

**Comment by james-prado** [comment]
Stale

**Pull request closed without merging**